cluster/plan.go

```diff
@@ -953,12 +951,32 @@ func (c *Cluster) BuildEtcdProcess(host *hosts.Host, etcdHosts []*hosts.Host, pr
 		fmt.Sprintf("%s:/etc/kubernetes:z", path.Join(prefixPath, "/etc/kubernetes")),
 	}
 
+	serviceOptions := c.GetKubernetesServicesOptions(host.DockerInfo.OSType, svcOptionData)
+	if serviceOptions.Etcd != nil {
+		for k, v := range serviceOptions.Etcd {
+			// if the value is empty, we remove that option
+			if len(v) == 0 {
+				delete(CommandArgs, k)
+				continue
+			}
+			CommandArgs[k] = v
+		}
+	}
+
 	for arg, value := range c.Services.Etcd.ExtraArgs {
 		if _, ok := c.Services.Etcd.ExtraArgs[arg]; ok {
 			CommandArgs[arg] = value
 		}
 	}
 
+	// adding the old default value from L922 if not present in metadata options or passed by user
+	if _, ok := CommandArgs["client-cert-auth"]; !ok {
+		args = append(args, "--client-cert-auth")
+	}
+	if _, ok := CommandArgs["peer-client-cert-auth"]; !ok {
+		args = append(args, "--peer-client-cert-auth")
```

Comment:
Same - `--peer-client-cert-auth=true`?

Reply:
this is to preserve the old behavior for existing clusters, otherwise they'll get auto upgraded and we don't want that